### ISSUE TITLE
Md5sum checks rule

### ISSUE THREAD

**Issue by jhagberg** [UPPMAX/irods#25]
Issue to extend the clear cache rule with md5sum verification.

**Comment by jhagberg** [comment]
Looks like the msiDataObjChksum behavior is not congruent. 

The option ChksumAll is not detecting a file on sweStore that has wrong checksum if correct replica on cache exist. I will ask the iRODS-chat about it.  (https://groups.google.com/d/msg/irod-chat/BvB_qezuPUo/CMF32Pzz8c4J)

We should consider in the module file to alias iput to iput -K 
-K flag is calculating the md5sum before and after first transfer to iRODS. 

The replication to sweStore also have option to verify the transfer with checksum. I guess this should be put in core.re

**Comment by jhagberg** [comment]
Since version 3.1 

"Compound object checksums. A checksum computation for compound class objects was added. This is done by staging data-objects to cache and then computing the checksum of the staged object"

**Comment by jhagberg** [comment]
After discussion with @samuell we came up with the idea to do some sort of verification in the univMSSdriver.

I found in the latest arc client documentation that you actually can provide an md5sum to the srm url that will work as a verification of the transfer and if the md5sum is not correct the transfer will fail. (http://www.nordugrid.org/documents/arc-ui.pdf) 

I have updated the univmssdriver to get the registered md5sum in iCAT of the file if not calculate the md5sum of the file. 
There is no option to chksum other users files as admin. But there might be a work around. 

https://groups.google.com/d/msg/iROD-Chat/dWuZUBqZaNU/1DKxp2y-TRIJ

If users always use iput -K the transfer to cache will be verified and the md5sum will be stored in iRODS iCAT database and the user provided md5sum will be used to also verify transfer to sweStore.
